test/extensions/filters/http/ext_proc/ext_proc_full_duplex_integration_test.cc

```diff
@@ -447,6 +447,39 @@ TEST_P(ExtProcIntegrationTest, NoneToFullDuplexMoreDataAfterModeOverride) {
   verifyDownstreamResponse(*response, 200);
 }
 
+TEST_P(ExtProcIntegrationTest, ServerWaitforEnvoyHalfCloseThenCloseStream) {
```

Comment:
can we have a parameterized test that use enumerate through the response types and possible "shortcut switches" (i.e. eos-in-mxn-mode, CONTINUE_AND_REPLACE in header response etc) and check whether an early half-close is sent? 
I know this is a lot of cases, but with this set up, we will have confidence in the changes we are making in this PR and upcoming PRs.

Reply:
Please check the changes in filter_test.cc and filter_full_duplex_test.cc.  The config_->stats().streams_closed_ counters are specially checked to make sure we are now closing the streams ASAP in those cases.